functions/go/inflate-helm-chart/main.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/inflate-helm-chart/third_party/sigs.k8s.io/kustomize/api/builtins"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+//nolint
+func main() {
+	asp := HelmChartProcessor{}
+	cmd := command.Build(&asp, command.StandaloneEnabled, false)
+
+	cmd.Short = "inflate helm chart"
+	cmd.Long = "inflate helm chart"
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+type HelmChartProcessor struct{}
+
+func (slp *HelmChartProcessor) Process(resourceList *framework.ResourceList) error {
+	err := run(resourceList)
+	if err != nil {
+		resourceList.Result = &framework.Result{
+			Name: "inflate-helm-chart",
+			Items: []framework.ResultItem{
+				{
+					Message:  err.Error(),
+					Severity: framework.Error,
+				},
+			},
+		}
+		return resourceList.Result
+	}
+	return nil
+}
+
+type helmChartInflatorFunction struct {
+	kyaml.ResourceMeta `json:",inline" yaml:",inline"`
+	plugins []builtins.Plugin
+}
+
+func (f *helmChartInflatorFunction) Config(rn *kyaml.RNode) error {
+	y, err := rn.String()
+	if err != nil {
+		return fmt.Errorf("cannot get YAML from RNode: %w", err)
+	}
+	kind, err := f.getKind(rn)
+	if err != nil {
+		return err
+	}
+	switch kind {
+	case "InflateHelmChart":
+		err = f.ConfigHelmArgs(nil, []byte(y))
+		if err != nil {
+			return err
+		}
+	case "ConfigMap":
+		dataMap := rn.GetDataMap()
+		bytes, err := kyaml.Marshal(dataMap)
+		if err != nil {
+			return err
+		}
+		err = f.ConfigMapArgs(bytes)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("`functionConfig` must be 'ConfigMap' or 'InflateHelmChart'")
```

Comment:
Please use `backquote` for ConfigMap and InflateHelmChart

Reply:
fixed